riscv/interactive.cc

```diff
@@ -774,3 +776,16 @@ void sim_t::interactive_until(const std::string& cmd, const std::vector<std::str
     step(1);
   }
 }
+
+void sim_t::interactive_dumpmems(const std::string& cmd, const std::vector<std::string>& args)
+{
+  for (unsigned i = 0; i < mems.size(); i++) {
+    std::stringstream mem_fname;
+    mem_fname << "mem.0x" << std::hex << mems[i].first << ".bin";
+
+    std::ofstream mem_file;
+    mem_file.open(mem_fname.str());
+    mems[i].second->dump(mem_file);
+    mem_file.close();
```

Comment:
Pass in the filename to the constructor, which will open the file. Then let the destructor do the close.